packages/local_auth/android/src/main/java/io/flutter/plugins/localauth/AuthenticationHelper.java

```diff
@@ -75,24 +73,38 @@
       Lifecycle lifecycle,
       FragmentActivity activity,
       MethodCall call,
-      AuthCompletionHandler completionHandler) {
+      AuthCompletionHandler completionHandler,
+      boolean allowCredentials) {
     this.lifecycle = lifecycle;
     this.activity = activity;
     this.completionHandler = completionHandler;
     this.call = call;
     this.isAuthSticky = call.argument("stickyAuth");
     this.uiThreadExecutor = new UiThreadExecutor();
-    this.promptInfo =
-        new BiometricPrompt.PromptInfo.Builder()
-            .setDescription((String) call.argument("localizedReason"))
-            .setTitle((String) call.argument("signInTitle"))
-            .setSubtitle((String) call.argument("fingerprintHint"))
-            .setNegativeButtonText((String) call.argument("cancelButton"))
-            .setConfirmationRequired((Boolean) call.argument("sensitiveTransaction"))
-            .build();
+
+    if (allowCredentials) {
+      this.promptInfo =
+          new BiometricPrompt.PromptInfo.Builder()
+              .setDescription((String) call.argument("localizedReason"))
+              .setTitle((String) call.argument("signInTitle"))
+              .setSubtitle((String) call.argument("biometricHint"))
+              .setConfirmationRequired((Boolean) call.argument("sensitiveTransaction"))
+              .setDeviceCredentialAllowed(true)
```

Comment:
You can pass  `allowCredentials` to  `setDeviceCredentialAllowed` call and get rid  of the conditional.

Reply:
There was an issue with setting both `setNegativeButtonText` and `setDeviceCredentialAllowed`. 


From android docs:
> Note that enabling this option replaces the negative button on the prompt with one that allows the user to authenticate with their device credential, making it an error to call setNegativeButton(java.lang.CharSequence, java.util.concurrent.Executor, android.content.DialogInterface.OnClickListener).

I will refactor to try to make it more obvious what is happening.

Reply:
updated